File: ui/src/api/recycle-bin.ts
import client from './client'
import { type PaginationParams, type PaginationResponse } from './pagination'

export interface RecycleBinApp {
  id: string
  slug: string
  name: string
  description: string
  env_id: string
  env_name: string
  project_id: string
  project_name: string
  app_type: string
  container_image: string
  deleted_at: string
}

export interface RecycleBinEnv {
  id: string
  slug: string
  name: string
  description: string
  project_id: string
  project_name: string
  cluster_id: string
  cluster_name: string
  cluster_namespace: string
  deleted_at: string
}

export interface EnvDeletionConflict {
  apps: RecycleBinApp[]
}

export const recycleBinApi = {
  listApps: async (projectId?: string, params?: PaginationParams) => {
    return client.get('/v1/recycle-bin/apps', {
      params: { ...params, project_id: projectId }
    }) as Promise<{ items: RecycleBinApp[], pagination: PaginationResponse }>
  },

  listEnvs: async (projectId?: string, params?: PaginationParams) => {
    return client.get('/v1/recycle-bin/envs', {
      params: { ...params, project_id: projectId }
    }) as Promise<{ items: RecycleBinEnv[], pagination: PaginationResponse }>
  },

  restoreApps: async (ids: string[]) => {
    return client.post('/v1/recycle-bin/apps/restore', { ids })
  },

  permanentlyDeleteApps: async (ids: string[]) => {
    return client.post('/v1/recycle-bin/apps/permanently-delete', { ids })
  },

  restoreEnvs: async (ids: string[]) => {
    return client.post('/v1/recycle-bin/envs/restore', { ids })
  },

  permanentlyDeleteEnvs: async (ids: string[]) => {
    return client.post('/v1/recycle-bin/envs/permanently-delete', { ids })
  },

  checkEnvDeletionConflicts: async (envId: string) => {
    return client.get(`/v1/recycle-bin/envs/${envId}/deletion-conflicts`) as Promise<EnvDeletionConflict>
  }
}

File: internal/models/recycle_bin.go
package models

import "time"

type RecycleBinAppResponse struct {
	ID             string    `json:"id"`
	Slug           string    `json:"slug"`
	Name           string    `json:"name"`
	Description    string    `json:"description"`
	EnvID          string    `json:"env_id"`
	EnvName        string    `json:"env_name"`
	ProjectID      string    `json:"project_id"`
	ProjectName    string    `json:"project_name"`
	ProjectSlug    string    `json:"project_slug"`
	AppType        string    `json:"app_type"`
	ContainerImage string    `json:"container_image"`
	DeletedAt      time.Time `json:"deleted_at"`
}

type ListRecycleBinAppResponse struct {
	Items      []RecycleBinAppResponse `json:"items"`
	Pagination PaginationResponse      `json:"pagination"`
}

type RecycleBinEnvResponse struct {
	ID               string    `json:"id"`
	Slug             string    `json:"slug"`
	Name             string    `json:"name"`
	Description      string    `json:"description"`
	ProjectID        string    `json:"project_id"`
	ProjectName      string    `json:"project_name"`
	ProjectSlug      string    `json:"project_slug"`
	ClusterID        string    `json:"cluster_id"`
	ClusterName      string    `json:"cluster_name"`
	ClusterNamespace string    `json:"cluster_namespace"`
	DeletedAt        time.Time `json:"deleted_at"`
}

type ListRecycleBinEnvResponse struct {
	Items      []RecycleBinEnvResponse `json:"items"`
	Pagination PaginationResponse      `json:"pagination"`
}

type RecycleBinProjectResponse struct {
	ID          string    `json:"id"`
	Slug        string    `json:"slug"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	DeletedAt   time.Time `json:"deleted_at"`
}

type ListRecycleBinProjectResponse struct {
	Items      []RecycleBinProjectResponse `json:"items"`
	Pagination PaginationResponse          `json:"pagination"`
}

type RestoreResourceRequest struct {
	IDs []string `json:"ids" binding:"required"`
}

type PermanentlyDeleteResourceRequest struct {
	IDs []string `json:"ids" binding:"required"`
}

type EnvDeletionConflictResponse struct {
	Apps []RecycleBinAppResponse `json:"apps"`
}
